### PR TITLE
burnitnow: fix crash at launch on x86_64

### LINKED ISSUE #1737: BurnItNow crashes on launch
The revision of BurnItNow found in HaikuDepot for x86_64 crashes on launch.
Below is the debugger report:

Debug information for team /boot/system/apps/BurnItNow (903):
CPU(s): 4x Intel Core™ i5-3570
Memory: 7.95 GiB total, 405.48 MiB used
Haiku revision: hrev51497 Oct 23 2017 11:07:02 (x86_64)

Active Threads:
	thread 904: team 903 debug task 
	thread 903: BurnItNow (main)
		state: Exception (Segment violation)

		Frame		IP			Function Name
		-----------------------------------------------
		0x7fd71626cb00	0x220b07e9ee	_GLOBAL(int16, , bool, , , , , , int32, int64, ) + 0xe 
			Disassembly:
				_GLOBAL(int16, , bool, , , , , , int32, int64, ):
				0x000000220b07e9e0:   488b0511222100  mov 0x212211(%rip), %rax
				0x000000220b07e9e7:               55  push %rbp
				0x000000220b07e9e8:           4889e5  mov %rsp, %rbp
				0x000000220b07e9eb:           488b38  mov (%rax), %rdi
				0x000000220b07e9ee:           488b07  mov (%rdi), %rax <--

			Frame memory:
				[0x7fd71626caf0]  ..&........."...   10 cb 26 16 d7 7f 00 00 c6 a7 08 0b 22 00 00 00
		0x7fd71626cb10	0x220b08a7c6	/boot/system/apps/BurnItNow + 0x217c6 
		0x7fd71626cb30	0x220b07cf20	/boot/system/apps/BurnItNow + 0x13f20 
		0x7fd71626cb80	0x1dbaee01fc7	init_dependencies(image_t*, bool) + 0xbd 
		0x7fd71626cbe0	0x1dbaee025e6	load_program + 0xc1 
		0x7fd71626cc10	0x1dbaee0ae0a	runtime_loader + 0xfd 
		00000000	0x7f8fae4ed260	commpage_thread_exit + 0 

		Registers:
			  rip:	0x000000220b07e9ee
			  rsp:	0x00007fd71626caf0
			  rbp:	0x00007fd71626caf0
			  rax:	0x000001d41ac08990
			  rbx:	0x000000220b28f028
			  rcx:	0x0000000000000000
			  rdx:	0x0000000000000000
			  rsi:	0x7fffffffffffffff
			  rdi:	0x0000000000000000
			   r8:	0x000000000000001c
			   r9:	0x00007fd71626caf8
			  r10:	0x000000c85789af7c
			  r11:	0x0000000000003202
			  r12:	0x0000000000000008
			  r13:	0x000000000000000f
			  r14:	0x0000000000000010
			  r15:	0x0000000000000008
			   cs:	0x0023
			   ds:	0x0000
			   es:	0x0000
			   fs:	0x0000
			   gs:	0x0000
			   ss:	0x001b
			  st0:	0
			  st1:	0
			  st2:	0
			  st3:	0
			  st4:	0
			  st5:	0
			  st6:	0
			  st7:	0
			  mm0:	{0, 0, 0, 0}
			  mm1:	{0, 0, 0, 0}
			  mm2:	{0, 0, 0, 0}
			  mm3:	{0, 0, 0, 0}
			  mm4:	{0, 0, 0, 0}
			  mm5:	{0, 0, 0, 0}
			  mm6:	{0, 0, 0, 0}
			  mm7:	{0, 0, 0, 0}
			 xmm0:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm1:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm2:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm3:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm4:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm5:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm6:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm7:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm8:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm9:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm10:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm11:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm12:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm13:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm14:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm15:	{0, 0, 0, 0, 0, 0, 0, 0}

Loaded Images:
	ID		Text Base	Text End	Data Base	Data End	Type	Name
	--------------------------------------------------------------------------------
	2481	0x1ece735000	0x1ece9b2000	0x1ecebb1000	0x1ecebc6000	lib    	/boot/system/lib/libicui18n.so.57.1
	2474	0x220b069000	0x220b08f000	0x220b28f000	0x220b292000	app    	/boot/system/apps/BurnItNow
	2480	0xc23da89000	0xc23f307000	0xc23f506000	0xc23f507000	lib    	/boot/system/lib/libicudata.so.57.1
	2479	0xc85785c000	0xc8579b3000	0xc857bb3000	0xc857c06000	lib    	/boot/system/lib/libroot.so
	2488	0x10627f93000	0x10627fa8000	0x106281a7000	0x106281a9000	lib    	/boot/system/lib/libtranslation.so
	2487	0x106ffaa1000	0x106ffabd000	0x106ffcbc000	0x106ffcbd000	lib    	/boot/system/lib/libz.so.1.2.11
	2478	0x1108cf48000	0x1108cf5d000	0x1108d15c000	0x1108d15d000	lib    	/boot/system/lib/libgcc_s.so.1
	2477	0x11337f01000	0x11338063000	0x11338263000	0x11338277000	lib    	/boot/system/lib/libstdc++.so.6.0.21
	2486	0x172faa63000	0x172fabfb000	0x172fadfa000	0x172fae0f000	lib    	/boot/system/lib/libicuuc.so.57.1
	2489	0x1b4fe302000	0x1b4fe3fe000	0x1b4fe5fe000	0x1b4fe601000	lib    	/boot/system/lib/libtextencoding.so
	2485	0x1b51b32d000	0x1b51b35c000	0x1b51b55c000	0x1b51b590000	lib    	/boot/system/lib/libicutu.so.57.1
	2482	0x1b62682c000	0x1b626838000	0x1b626a37000	0x1b626a39000	lib    	/boot/system/lib/libicuio.so.57.1
	2484	0x1d0adf8a000	0x1d0adf95000	0x1d0ae195000	0x1d0ae196000	lib    	/boot/system/lib/libiculx.so.57.1
	2475	0x1d41a6df000	0x1d41a9dc000	0x1d41abdb000	0x1d41ac09000	lib    	/boot/system/lib/libbe.so
	2473	0x1dbaedf7000	0x1dbaee15000	0x1dbaf014000	0x1dbaf016000	lib    	/boot/system/runtime_loader
	2483	0x244f95db000	0x244f9630000	0x244f9830000	0x244f9833000	lib    	/boot/system/lib/libicule.so.57.1
	2476	0x25841d88000	0x25841f20000	0x25842120000	0x25842139000	lib    	/boot/system/lib/libtracker.so
	2472	0x7f8fae4ed000	0x7f8fae4f5000	0x00000000	0x00000000	system 	commpage

Areas:
	ID		Base		End			Size (KiB)	Protection	Locking			Name
	--------------------------------------------------------------------------------
	7560	0x15ef53000	0x15ef6b000	        96	rw-        	full          	rld heap
	7559	0x347935000	0x34794d000	        96	rw-        	full          	rld heap
	7558	0x61c584000	0x61c594000	        64	rw-        	full          	rld heap
	7557	0x683215000	0x683225000	        64	rw-        	full          	rld heap
	7556	0xef21d1000	0xef21d9000	        32	rw-        	full          	rld heap
	7537	0x1ece735000	0x1ece9b2000	      2548	r-x        	full          	libicui18n.so.57.1_seg0ro
	7538	0x1ecebb1000	0x1ecebc6000	        84	rw-        	full          	libicui18n.so.57.1_seg1rw
	7520	0x220b069000	0x220b08f000	       152	r-x        	full          	BurnItNow_seg0ro
	7521	0x220b28f000	0x220b292000	        12	rw-        	full          	BurnItNow_seg1rw
	7519	0x5b45f72000	0x5b45f73000	         4	rw-S       	full          	_rld_debug_
	7518	0x5d998dc000	0x5d998ec000	        64	rw-        	full          	rld heap
	7561	0x8fd79de000	0x8fd7a8e000	       704	rw-        	full          	heap
	7535	0xc23da89000	0xc23f307000	     25080	r-x        	full          	libicudata.so.57.1_seg0ro
	7536	0xc23f506000	0xc23f507000	         4	rw-        	full          	libicudata.so.57.1_seg1rw
	7532	0xc85785c000	0xc8579b3000	      1372	r-x        	full          	libroot.so_seg0ro
	7533	0xc857bb3000	0xc857bc1000	        56	rw-        	full          	libroot.so_seg1rw
	7534	0xc857bc1000	0xc857c06000	       276	rw-        	full          	libroot.so_seg2rw
	7552	0x10627f93000	0x10627fa8000	        84	r-x        	full          	libtranslation.so_seg0ro
	7553	0x106281a7000	0x106281a9000	         8	rw-        	full          	libtranslation.so_seg1rw
	7550	0x106ffaa1000	0x106ffabd000	       112	r-x        	full          	libz.so.1.2.11_seg0ro
	7551	0x106ffcbc000	0x106ffcbd000	         4	rw-        	full          	libz.so.1.2.11_seg1rw
	7530	0x1108cf48000	0x1108cf5d000	        84	r-x        	full          	libgcc_s.so.1_seg0ro
	7531	0x1108d15c000	0x1108d15d000	         4	rw-        	full          	libgcc_s.so.1_seg1rw
	7527	0x11337f01000	0x11338063000	      1416	r-x        	full          	libstdc++.so.6.0.21_seg0ro
	7528	0x11338263000	0x1133826f000	        48	rw-        	full          	libstdc++.so.6.0.21_seg1rw
	7529	0x1133826f000	0x11338277000	        32	rw-        	full          	libstdc++.so.6.0.21_seg2rw
	7548	0x172faa63000	0x172fabfb000	      1632	r-x        	full          	libicuuc.so.57.1_seg0ro
	7549	0x172fadfa000	0x172fae0f000	        84	rw-        	full          	libicuuc.so.57.1_seg1rw
	7554	0x1b4fe302000	0x1b4fe3fe000	      1008	r-x        	full          	libtextencoding.so_seg0ro
	7555	0x1b4fe5fe000	0x1b4fe601000	        12	rw-        	full          	libtextencoding.so_seg1rw
	7545	0x1b51b32d000	0x1b51b35c000	       188	r-x        	full          	libicutu.so.57.1_seg0ro
	7546	0x1b51b55c000	0x1b51b55e000	         8	rw-        	full          	libicutu.so.57.1_seg1rw
	7547	0x1b51b55e000	0x1b51b590000	       200	rw-        	full          	libicutu.so.57.1_seg2rw
	7539	0x1b62682c000	0x1b626838000	        48	r-x        	full          	libicuio.so.57.1_seg0ro
	7540	0x1b626a37000	0x1b626a39000	         8	rw-        	full          	libicuio.so.57.1_seg1rw
	7543	0x1d0adf8a000	0x1d0adf95000	        44	r-x        	full          	libiculx.so.57.1_seg0ro
	7544	0x1d0ae195000	0x1d0ae196000	         4	rw-        	full          	libiculx.so.57.1_seg1rw
	7522	0x1d41a6df000	0x1d41a9dc000	      3060	r-x        	full          	libbe.so_seg0ro
	7523	0x1d41abdb000	0x1d41ac09000	       184	rw-        	full          	libbe.so_seg1rw
	7516	0x1dbaedf7000	0x1dbaee15000	       120	r-x        	full          	runtime_loader_seg0ro
	7517	0x1dbaf014000	0x1dbaf016000	         8	rw-        	full          	runtime_loader_seg1rw
	7541	0x244f95db000	0x244f9630000	       340	r-x        	full          	libicule.so.57.1_seg0ro
	7542	0x244f9830000	0x244f9833000	        12	rw-        	full          	libicule.so.57.1_seg1rw
	7524	0x25841d88000	0x25841f20000	      1632	r-x        	full          	libtracker.so_seg0ro
	7525	0x25842120000	0x25842138000	        96	rw-        	full          	libtracker.so_seg1rw
	7526	0x25842138000	0x25842139000	         4	rw-        	full          	libtracker.so_seg2rw
	7515	0x7f8fae4ed000	0x7f8fae4f5000	        32	r-xSk      	full          	commpage
	7514	0x7fd71526d000	0x7fd71626e000	     16388	rw-s       	full          	BurnItNow_903_stack
	7512	0x7ff7732e1000	0x7ff7732e5000	        16	rw-        	full          	user area

Protection Flags: r - read, w - write, x - execute, s - stack, o - overcommit, c - cloneable, S - shared, k - kernel

Semaphores:
	ID		Count	Last Holder	Name
	------------------------------------------------------------
	4909	    0	          0	some BBlockCache lock
	4910	    0	          0	Catalog
	4911	    0	          0	some BLocker
	4912	    0	          0	some BLocker
	4913	    0	          0	token space
	4914	    0	          0	BLooperList lock
	4915	    0	          0	AppServerLink_sLock
	4916	    0	          0	LocaleRosterData
	4917	    0	          0	some BLocker
	4918	    0	          0	Catalog
	4919	    0	          0	PeriodicUpdatePoses
	4920	    0	          0	Light Lock
	4921	    0	          0	Light Lock
	4922	    0	          0	undo
	4923	    0	          0	some BLocker
	4924	    0	          0	some BLocker

### PR DESCRIPTION
**Pull request by phoudoin**
Fix #1737 until a new BurnItNow release including its own fix is available...

### CODE REVIEW

File: haiku-apps/burnitnow/burnitnow-1.1.recipe [hunk 1]
```diff
@@ -25,12 +26,12 @@ REQUIRES="
 	"
 
 BUILD_REQUIRES="
-	haiku_devel
+	haiku${secondaryArchSuffix}_devel
```
Comment (korli) [inline haiku-apps/burnitnow/burnitnow-1.1.recipe]: This change isn't needed, there no secondary arch for this app.

File: haiku-apps/burnitnow/burnitnow-1.1.recipe [hunk 2]
```diff
@@ -25,12 +26,12 @@ REQUIRES="
...
 	"
 BUILD_PREREQUIRES="
 	makefile_engine
-	cmd:g++
-	cmd:ld
+	cmd:g++$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
```
Comment (korli) [inline haiku-apps/burnitnow/burnitnow-1.1.recipe]: Ditto.

**Comment by scottmc** [comment]
Can you also open a pull request for the patch to be added to BurnItNow on HaikuArchives?  https://github.com/HaikuArchives/BurnItNow
We should try to avoid adding patches to Haiku apps that we have the source code to and are hosted on GitHub, and just fix it at the source.

**Comment by korli** [comment]
@scottmc in this case upstream has already a fix, but mixed in other changes.

File: haiku-apps/burnitnow/burnitnow-1.1.recipe (hunk 1 above)
Comment (phoudoin) [reply haiku-apps/burnitnow/burnitnow-1.1.recipe]: Damned. I'm lost on when it's needed and when it's not...

At least it doesn't harm here. 
Will fix it anyway.

**Comment by phoudoin** [comment]
@scottmc a fix is already in burnitnow master branch, but it wasn't backported in a release. 
Hence this (temporary) patch on the latest release available.

**Pull request merged**